src/ServiceControl.Monitoring/Infrastructure/VariableHistoryIntervalStore.cs

```diff
@@ -0,0 +1,45 @@
+﻿namespace ServiceControl.Monitoring.Infrastructure
+{
+    using System;
+    using System.Collections.Concurrent;
+    using Messaging;
+
+    public abstract class VariableHistoryIntervalStore
+    {
+        ConcurrentDictionary<HistoryPeriod, IntervalsStore> histories;
```

Comment:
We've got `HistoryPeriod.All`, so this could be made a regular `Dictionary`. The stores underneath are mutable and thread safe so there's no gain in using a `ConcurrentDict` in here.